lib/mixlib/config.rb

```diff
@@ -457,9 +457,18 @@ def define_attr_accessor_methods(symbol)
         internal_set(symbol, value)
       end
       # Getter
-      meta.send :define_method, symbol do |*args|
-        internal_get_or_set(symbol, *args)
+      meta.send :define_method, symbol do |*args, &block|
+        # If a block was given, eval it in the context
+        if block
+          context_eval(symbol, &block)
+        else
+          internal_get_or_set(symbol, *args)
+        end
       end
     end
+
+    def context_eval(context, &block)
```

Comment:
not really sure this method adds much at all, i'd prefer to just see the code line there